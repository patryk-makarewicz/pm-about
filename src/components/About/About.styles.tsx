import styled, { css } from 'styled-components';

import { colors } from 'styles/colors';
import { fontSize, fontWeight, margin, shadow } from 'styles/variables';

type TextProps = {
  $marginBottom?: string;
};

type PhotoProps = {
  hide: boolean;
};

export const Wrapper = styled.section``;

export const Box = styled.div`
  width: 120px;
  height: 120px;
  border-radius: 50%;
  border: 4px solid var(--background-color);
  box-shadow: ${shadow.first};
  margin-bottom: ${margin.l};
  overflow: hidden;
`;

export const Photo = styled.img<PhotoProps>`
  width: 100%;
  height: 100%;
  transition: ease 1s all;
  ${({ hide }) =>
    hide
      ? css`
          opacity: 0;
        `
      : css`
          opacity: 1;
        `}
  z-index: 1
`;

export const Greeting = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const Cloud = styled.div`
  position: relative;
  background-color: var(--background-color);
  color: var(--text-color);
  border-radius: 8px;
  padding: 10px;
  font-weight: ${fontWeight.semiBold};
  box-shadow: ${shadow.first};
  margin-bottom: 10px;
  margin-left: 120px;

  &::after {
    content: '';
    position: absolute;
    top: 100%;
    left: 60%;
    margin-left: -50px;
    width: 0;
    height: 0;
    border-top: solid 5px var(--background-color);
    border-left: solid 5px transparent;
    border-right: solid 5px transparent;
  }
`;

export const Hand = styled.span`
  @keyframes wave {
    0% {
      transform: rotate(-20deg);
    }
    50% {
      transform: rotate(30deg);
    }
    100% {
      transform: rotate(-10deg);
    }
  }
  display: inline-block;
  animation: wave 1.5s ease-in-out infinite;
  transform-origin: center center;
`;

export const Text = styled.p<TextProps>`
  color: var(--text-color);
  font-size: ${fontSize.m};
  margin: 0 auto;
  ${({ $marginBottom }) =>
    $marginBottom &&
    css`
      margin-bottom: ${$marginBottom};
    `}
  max-width: 500px;
  text-align: center;
`;

export const TextGreen = styled.span`
  color: ${colors.green};
  font-weight: ${fontWeight.bold};
`;

export const ButtonIcon = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 44px;
  height: 44px;
  background-color: var(--background-color);
  border-radius: 50%;
  box-shadow: ${shadow.first};
  transition: all 0.2s ease-in-out;

  :hover {
    box-shadow: 0 2px 6px rgba(0, 0, 0, 0.2);
    opacity: 0.8;
  }
`;
