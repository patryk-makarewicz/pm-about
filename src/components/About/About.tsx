import { useTranslation } from 'react-i18next';

import { Placeholder } from 'components/Placeholder';
import { SocialMedia } from 'components/SocialMedia/SocialMedia';

import { PhotoLoad } from 'utils/photo';

import Me from '../../assets/me.jpg';
import * as Styled from './About.styles';

export const About = () => {
  const { t } = useTranslation();

  const { onLoad, loaded, refImage } = PhotoLoad();

  return (
    <Styled.Wrapper>
      <Styled.Greeting>
        <Styled.Cloud>
          {t('greeting.hi')} <Styled.Hand>🖐</Styled.Hand>
        </Styled.Cloud>
        <Styled.Box>
          <Placeholder hide={loaded} />
          <Styled.Photo src={Me} alt="Me" title="Patryk Makarewicz" ref={refImage} onLoad={onLoad} hide={!loaded} />
        </Styled.Box>
      </Styled.Greeting>
      <Styled.Text $marginBottom="10px">
        {t('greeting.dev')} <Styled.TextGreen>{t('greeting.devGreen')}</Styled.TextGreen>
      </Styled.Text>
      <Styled.Text $marginBottom="20px">{t('greeting.content')} 😊.</Styled.Text>
      <SocialMedia />
    </Styled.Wrapper>
  );
};
