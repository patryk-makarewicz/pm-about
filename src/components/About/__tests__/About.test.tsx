import { render, cleanup } from '@testing-library/react';

import { About } from '../About';

afterEach(cleanup);

const renderAbout = () => {
  const { baseElement } = render(<About />);

  return {
    baseElement
  };
};

describe('Testing About Component', () => {
  test('should take snapshot', async () => {
    const { baseElement } = renderAbout();

    expect(baseElement).toMatchSnapshot();
  });
});
