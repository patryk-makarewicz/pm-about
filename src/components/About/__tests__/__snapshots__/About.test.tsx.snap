// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing About Component should take snapshot 1`] = `
.c3 {
  width: 120px;
  height: 120px;
  border-radius: 50%;
  border: 4px solid var(--background-color);
  box-shadow: 0 4px 12px rgba(0,0,0,0.2);
  margin-bottom: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  position: relative;
  background-color: var(--background-color);
  color: var(--text-color);
  border-radius: 8px;
  padding: 10px;
  font-weight: 600;
  box-shadow: 0 4px 12px rgba(0,0,0,0.2);
  margin-bottom: 10px;
  margin-left: 120px;
}

.c1::after {
  content: '';
  position: absolute;
  top: 100%;
  left: 60%;
  margin-left: -50px;
  width: 0;
  height: 0;
  border-top: solid 5px var(--background-color);
  border-left: solid 5px transparent;
  border-right: solid 5px transparent;
}

.c2 {
  display: inline-block;
  -webkit-animation: wave 1.5s ease-in-out infinite;
  animation: wave 1.5s ease-in-out infinite;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
}

.c4 {
  color: var(--text-color);
  font-size: 1rem;
  margin: 0 auto;
  margin-bottom: 10px;
  max-width: 500px;
  text-align: center;
}

.c6 {
  color: var(--text-color);
  font-size: 1rem;
  margin: 0 auto;
  margin-bottom: 20px;
  max-width: 500px;
  text-align: center;
}

.c5 {
  color: #389e0d;
  font-weight: 700;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 44px;
  height: 44px;
  background-color: var(--background-color);
  border-radius: 50%;
  box-shadow: 0 4px 12px rgba(0,0,0,0.2);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7:hover {
  box-shadow: 0 2px 6px rgba(0,0,0,0.2);
  opacity: 0.8;
}

<body>
  <div>
    <section
      class=""
    >
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          greeting.hi
           
          <span
            class="c2"
          >
            üñê
          </span>
        </div>
        <img
          alt="Me"
          class="c3"
          src="me.jpg"
          title="Patryk Makarewicz"
        />
      </div>
      <p
        class="c4"
      >
        greeting.dev
         
        <span
          class="c5"
        >
          greeting.devGreen
        </span>
      </p>
      <p
        class="c6"
      >
        greeting.content
         üòä.
      </p>
      <div
        class="ant-row ant-row-center"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <a
              class="c7"
              href="https://www.linkedin.com/in/patryk-makarewicz/"
              target="_blank"
              title="Go to LinkedIn"
            >
              <svg
                height="36"
                width="36"
              >
                linkedin.svg
              </svg>
            </a>
          </div>
          <div
            class="ant-space-item"
          >
            <a
              class="c7"
              href="https://github.com/patryk-makarewicz"
              target="_blank"
              title="Go to Github"
            >
              <svg
                height="36"
                width="36"
              >
                github.svg
              </svg>
            </a>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
